lib/grandpa/grandpa.go

```diff
@@ -550,6 +553,8 @@ func (s *Service) sendVoteMessage(stage Subround, msg *VoteMessage, roundComplet
 	ticker := time.NewTicker(s.interval * 4)
 	defer ticker.Stop()
 
+	// Eventhough, this looks like we are sending messages multiple times,
+	// caching would make sure that they are being sent only once.
```

Comment:
Is it caching then? Maybe rename it to something like recorder? :thinking:

Reply:
> Is it caching then? Maybe rename it to something like recorder?

You not wrong, but we have called this caching everywhere (outside this package as well, where we need to store messages). A lot of variable renaming.